### PR TITLE
Add blog spot for Spring native images

### PR DESCRIPTION
**Pull request by jeanbisutti**
---

**Preview**:  https://deploy-preview-3612--opentelemetry.netlify.app/blog/2023/spring-native/

### CODE REVIEW

File: content/en/blog/2023/spring-native/index.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: Observe your Spring native application with OpenTelemetry instrumentation
+linkTitle: OTel Spring native
+date: 2023-11-28
+author: >-
+[Jean Bisutti](https://github.com/jeanbisutti) (Microsoft)
+---
```
Comment (cartermp) [inline content/en/blog/2023/spring-native/index.html:7]:
```suggestion
author: >-
  [Jean Bisutti](https://github.com/jeanbisutti) (Microsoft)
---
```

**Comment by cartermp** [comment]
I think the above should fix the yaml parsing error that netlify threw

**Comment by cartermp** [comment]
Looks like a refcache update is needed, if you type `/fix:refcache` the bot should take care of it

**Comment by jeanbisutti** [comment]
> /fix:refcache

It seems that the `/fix:refcache` comment was not processed by the bot.

File: content/en/blog/2023/spring-native/index.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+title: Observe your Spring native application with OpenTelemetry instrumentation
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.html:2]:
```suggestion
title: Observe your Spring Native application with OpenTelemetry instrumentation
```

File: content/en/blog/2023/spring-native/index.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+linkTitle: OTel Spring native
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.html:3]:
```suggestion
linkTitle: OTel Spring Native
```

File: content/en/blog/2023/spring-native/index.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+date: 2023-11-28
+author: >-
+  [Jean Bisutti](https://github.com/jeanbisutti) (Microsoft)
+---
+
+{{< blog/integration-badge >}} You have just built a [Spring Boot native image
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.html:9]:
```suggestion
You have just built a [Spring Boot Native image
```

**Review by svrnm — CHANGES_REQUESTED** [review]

**Review by svrnm — APPROVED** [review]

**Review by chalin — DISMISSED** [review]
@jeanbisutti have you checked the preview https://deploy-preview-3612--opentelemetry.netlify.app/blog/2023/spring-native/?

Your markdown is broken, please fix it:

> image

**Comment by svrnm** [comment]
hey @jeanbisutti, 

after reading through this blog post once again and thinking about it for a little bit, I think it is too sparse to qualify for a blog post. While we had short & sparse blog posts in the past, they announced major updates to certain projects (e.g. .NET going stable, Go now having Metrics, etc.). This one points to a specific feature now being available for Java. We could tweet/announce it on social media that way, but to have it on the blog I would ask you to add a little bit more details (if possible), i.e. why is this exciting for the Java ecosystem? Are there more "starters" to come for other popular frameworks/native images and this is just the start? Is this the defacto way of doing otel for spring native images and other ways are not relevant anymore (like the agent), etc. 

Since I am not in the Java community, I would also like to hear some @open-telemetry/java-approvers opinion on if and how this should be presented?

**Comment by chalin** [comment]
The markdown is still broken (check the preview), but regardless, I agree with @svrnm that a social media post would be _much_ more appropriate for this.

**Comment by jeanbisutti** [comment]
I have reopened the PR with a new version.

**Comment by jeanbisutti** [comment]
Dynamic version number does not seem to be managed for blog posts?

`{{% param vers.instrumentation %}}-alpha`

**Comment by svrnm** [comment]
> Dynamic version number does not seem to be managed for blog posts?
> 
> `{{% param vers.instrumentation %}}-alpha`

No, with blog post fix the current version. Blog posts will outdate eventually, so this is not an issue.

File: content/en/blog/2023/spring-native/index.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
+---
+title: Observe your Spring Native application with OpenTelemetry instrumentation
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:2]:
```suggestion
title: Observe your Spring Native application with OpenTelemetry
```

File: content/en/blog/2023/spring-native/index.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+You may know the
+[OpenTelemetry Java agent](https://opentelemetry.io/docs/instrumentation/java/automatic/).
+However, Java agents are
+[not officially supported](https://github.com/oracle/graal/issues/1065) with
+GraalVM native images today. This post presents an OpenTelemetry solution to get
+telemetry data with Spring Boot native image applications. ## Instrumentation of
+Spring Boot native image applications - A history of the last months The
+OpenTelemetry contributors have used the existing OpenTelemetry Spring Boot
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:17]:
```suggestion
telemetry data with Spring Boot native image applications. 

## A history of the last months 

The OpenTelemetry contributors have used the existing OpenTelemetry Spring Boot
```

The section title is too long, not only for prettier but also eventually for the TOC, keep it short.

**Review by svrnm — CHANGES_REQUESTED** [review]
A few initial remarks, overall this is moving in the right direction. I will share some additional remarks later on

File: content/en/blog/2023/spring-native/index.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  [Jean Bisutti](https://github.com/jeanbisutti) (Microsoft)
+cSpell:ignore: datasource logback bisutti
+---
+
+You may know the
+[OpenTelemetry Java agent](https://opentelemetry.io/docs/instrumentation/java/automatic/).
+However, Java agents are
+[not officially supported](https://github.com/oracle/graal/issues/1065) with
+GraalVM native images today. This post presents an OpenTelemetry solution to get
+telemetry data with Spring Boot native image applications.
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:15]:
```suggestion
The [OpenTelemetry Java Agent]([/docs/instrumentation/java/automatic/](https://opentelemetry.io/docs/instrumentation/java/automatic/)) is a convenient and well-established way to instrument Java applications. However, as of today [it is not possible to use it with GraalVM Native Images](https://github.com/oracle/graal/issues/1065). 

To provide you with an easy and seamless way for Spring Boot Native Image application nevertheless, the OpenTelemetry Java contributors have improved the existing OpenTelemetry Spring Boot Starter to work well with Spring Boot Native Image applications. Read on to learn more!
```

A suggested re-wording of the first paragraph.

File: content/en/blog/2023/spring-native/index.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+```xml
+<dependencies>
+  <dependency>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-spring-boot-starter</artifactId>
+    <version>1.32.0-alpha</version>
+  </dependency>
+</dependencies>
+```
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:35]:
```suggestion
## Easy instrumention without an agent

The OpenTelemetry Spring Boot Starter allows to add OpenTelemetry to your application without byte code instrumentation. The OpenTelemetry Java contributors have used this to instrument Spring Boot Native Images. To make this experience better, they improved the Starter over the last few months: they have added the OTLP log export and an instrumentation library for Spring HTTP frameworks. 

By adding the starter dependency to your project, you will benefit from those two features out of the box:

```xml
<dependencies>
  <dependency>
    <groupId>io.opentelemetry.instrumentation</groupId>
    <artifactId>opentelemetry-spring-boot-starter</artifactId>
    <version>1.32.0-alpha</version>
  </dependency>
</dependencies>
```
```

File: content/en/blog/2023/spring-native/index.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+[Spring Boot native image applications do not support Log4j2 logging library](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-with-GraalVM).
+They have also worked to reduce the configuration to set up the logging and
+database instrumentation with the OpenTelemetry Starter. For example, if your
+application does not declare DataSource bean, you enable the database by simply
+adding two properties in the application.properties file:
+
+```properties
+spring.datasource.url=jdbc:otel:h2:mem:db
+spring.datasource.driver-class-name=io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver
+```
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:52]:
~~~suggestion
To get even more visibility, the Starter can be combined with [instrumentation libraries](/docs/instrumentation/java/libraries/). For this purpose, the OpenTelemetry Java contributors have improved the JDBC (database)
libraries and logging instrumentation libraries. For example, for the Logback logging library, they have added GraalVM configuration to make the library work in native mode[^1].

Furthermore, they have worked to reduce the configuration to set up the logging and
database instrumentation with the Starter. For example, if your
application does not declare a DataSource bean, you can enable the database by simply
adding two properties in the `application.properties` file:

```properties
spring.datasource.url=jdbc:otel:h2:mem:db
spring.datasource.driver-class-name=io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver
```
~~~

File: content/en/blog/2023/spring-native/index.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+Feel free to read the
+[documentation](https://opentelemetry.io/docs/instrumentation/java/automatic/spring-boot/)
+of the OpenTelemetry starter to learn more. You can also use
+[opentelemetry-java-examples/spring-native](https://github.com/open-telemetry/opentelemetry-java-examples/tree/main/spring-native)
+to run a Spring Boot native image application and look at the generated HTTP,
+log, and database telemetry data. Note that the OpenTelemetry developers have
+also added GraalVM native automatic tests to the OpenTelemetry Java
+instrumentation project to detect regressions related to the native mode
+execution.
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:62]:
```suggestion
Read the
[documentation](https://opentelemetry.io/docs/instrumentation/java/automatic/spring-boot/)
of the OpenTelemetry Spring Boot Starter to learn more. You can use
[opentelemetry-java-examples/spring-native](https://github.com/open-telemetry/opentelemetry-java-examples/tree/main/spring-native)
to run a Spring Boot Native Image application and look at the generated logs as well as HTTP and database telemetry data. 

Finally, the OpenTelemetry Java contributors have added GraalVM Native automatic tests to the OpenTelemetry Java project to detect regressions related to the native mode execution.
```

File: content/en/blog/2023/spring-native/index.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+## What’s next?
+
+The OpenTelemetry contributors expect to be able to configure the logging and
+database instrumentations automatically. Please share your thoughts and
+experience about the OpenTelemetry starter and Spring Boot native applications.
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:68]:
```suggestion
## What’s next?

The OpenTelemetry Java contributors expect to be able to enable automatic logging and
database instrumentation configuration. 

If you try out the OpenTelemetry Spring Boot Starter, share your thoughts and experiences via [GitHub discussions](https://github.com/open-telemetry/opentelemetry-java/discussions) or the [#otel-java](https://cloud-native.slack.com/archives/C014L2KCTE3) [CNCF Slack](https://slack.cncf.io) channel.

[^1]: Spring Boot Native Image applications [do not support Log4j2 logging library](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-with-GraalVM).
```

**Review by svrnm — CHANGES_REQUESTED** [review]
Some editing, please take a look

File: content/en/blog/2023/spring-native/index.md (hunk 8 above)
Comment (jeanbisutti) [reply content/en/blog/2023/spring-native/index.md]: It has changed the meaning. I have reworded in 6fe84919b2d150c110fec3a8dae60d4cfb077ccf What do youn think?

File: content/en/blog/2023/spring-native/index.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+title: Observe your Spring Native image application with OpenTelemetry
+linkTitle: Spring Native
+date: 2023-11-28
+author: >-
+  [Jean Bisutti](https://github.com/jeanbisutti) (Microsoft)
+cSpell:ignore: datasource logback bisutti
+---
+
+The
+[OpenTelemetry Java Agent](<[/docs/instrumentation/java/automatic/](https://opentelemetry.io/docs/instrumentation/java/automatic/)>)
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:11]:
```suggestion
[OpenTelemetry Java agent](<[/docs/instrumentation/java/automatic/](https://opentelemetry.io/docs/instrumentation/java/automatic/)>)
```

File: content/en/blog/2023/spring-native/index.md (hunk 8 above)
Comment (svrnm) [reply content/en/blog/2023/spring-native/index.md]: works for me!

File: content/en/blog/2023/spring-native/index.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+title: Observe your Spring Native Image application with OpenTelemetry
+linkTitle: Observe Spring Native
+date: 2023-12-07
+author: >-
+  [Jean Bisutti](https://github.com/jeanbisutti) (Microsoft)
+cSpell:ignore: datasource logback bisutti
+---
+
+The
+[OpenTelemetry Java agent](<[/docs/instrumentation/java/automatic/](https://opentelemetry.io/docs/instrumentation/java/automatic/)>)
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:11]:
```suggestion
[OpenTelemetry Java agent](/docs/instrumentation/java/automatic)
```

File: content/en/blog/2023/spring-native/index.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
+---
+title: Observe your Spring Native Image application with OpenTelemetry
+linkTitle: Observe Spring Native
+date: 2023-12-07
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:4]:
```suggestion
date: 2023-12-08
```

File: content/en/blog/2023/spring-native/index.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+does not declare a DataSource bean, you can enable the database by simply adding
+two properties in the `application.properties` file:
+
+```properties
+spring.datasource.url=jdbc:otel:h2:mem:db
+spring.datasource.driver-class-name=io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver
+```
+
+Read the
+[documentation](https://opentelemetry.io/docs/instrumentation/java/automatic/spring-boot/)
```
Comment (svrnm) [inline content/en/blog/2023/spring-native/index.md:58]:
```suggestion
[documentation](/docs/instrumentation/java/automatic/spring-boot/)
```

**Review by svrnm — APPROVED** [review]

**Pull request merged**